### PR TITLE
Add an Ops philosophy

### LINKED ISSUE #143: Create a "SciJava ops philosophy" document
ImageJ ops and SciJava ops clearly have distinct design philosophies that they abide by. While I know some of the reasons for why SciJava ops works the way it does, new users may be really mystified by it. Especially if they have any experience with ImageJ ops. For example when I was first interacting with SciJava ops, I thought the `ops.binary()` was a special namespace for _binary_ images. So this call: `ops.binary("filter.gauss").input(inImage, 2.0).output(outImage).computer()` had me a little confused for a moment.

I think it would be a great starting point for new and existing users (_i.e._ I'd probably would refer to it myself and share it).

### PR DESCRIPTION
**Pull request by gselzer**
This PR adds a new page to the documentation to describe the philosophy of Ops (i.e. "Why Ops").

I'm a bit wary of whether this document makes sense for the audience, so I'm looking for heavy critique from @ctrueden @hinerm @elevans before merging

Closes scijava/scijava#143

### CODE REVIEW

**Comment by elevans** [comment]
@gselzer So far I like the document and it helps me understand the "gap" that SciJava Ops will fill in the image algorithm space. What I would like to see added to the document at some point is some text that helps soften the blow of the (from my perspective) unusual syntax of a SciJava Op call (_i.e._ `ops.op("filter.gauss").arity2().input(Img, 3.0)`). Specifically, I'm expecting people to really be thrown off on why they need to specify the input arity for their Op call. Some text that says "this is why we do this and why its best" would be great!

**Comment by gselzer** [comment]
@elevans thank you for taking a look.

> What I would like to see added to the document at some point is some text that helps soften the blow of the (from my perspective) unusual syntax of a SciJava Op call (_i.e._ `ops.op("filter.gauss").arity2().input(Img, 3.0)`). Specifically, I'm expecting people to really be thrown off on why they need to specify the input arity for their Op call.

Yeah, that's needed, although we might argue that it is better to put that discussion elsewhere. Is there not something like that in the `Calling Ops` page of the documentation?

>  Some text that says "this is why we do this and why its best" would be great!

I don't think I can write this well, having been an [opponent](https://github.com/scijava/incubator/pull/74#issuecomment-1478357786) of this change. @hinerm or @ctrueden I would really appreciate if one of you could write this section because as proponents of this change you may be able to better convince users that it is worthwhile.

**Comment by hinerm** [comment]
>What I would like to see added to the document at some point is some text that helps soften the blow of the (from my perspective) unusual syntax of a SciJava Op call (i.e. ops.op("filter.gauss").arity2().input(Img, 3.0)

> I don't think I can write this well, having been an [opponent](https://github.com/scijava/incubator/pull/74#issuecomment-1478357786) of this change. @hinerm or @ctrueden I would really appreciate if one of you could write this section because as proponents of this change you may be able to better convince users that it is worthwhile.

You do not *need* to say `ops.op.arity` - you can write `ops.binary`. As I [said here](https://github.com/scijava/incubator/pull/74#issuecomment-1478482715), separating arity and input type to two steps in the builder gives you an additive number of choices to make, and you never see more than `[max arity]` options at a time, while combining them in a single step gives you a multiplicative number of choices.

And as @ctrueden [wrote](https://github.com/scijava/incubator/pull/74#issuecomment-1492391584), this API is Java-developer-focused (e.g. programming from an IDE). We can write convenience layers on top for python/scripting for users, as needed.

I vote against API-level documentation or explanation in this philosophy document.

File: docs/ops/doc/WhyOps.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+* an image and a shape over which to compute the gaussian blur
+* ...
+
+It is SciJava Ops' goal to collapse all algorithms into the following descriptions, and to provide a uniform execution pathway for each - hiding the complexities of each individual algorithm library.
+
+## Isn't SciJava Ops just another algorithm library?
+
+While it may currently seem that way, SciJava Ops is designed to *wrap* existing algorithms, not to create new ones (although you can certainly do that!)
+
+To that end, SciJava Ops
```
Comment (hinerm) [inline docs/ops/doc/WhyOps.md]: @gselzer did you have more written that goes here? Do you have more sections planned?

Comment (gselzer) [reply docs/ops/doc/WhyOps.md]: @hinerm I just wanted to write something along the lines of "SciJava exposes a rich API for declaring Ops from arbitrary sources, spanning libraries and programming languages". Sorry I forgot!

**Review by hinerm — APPROVED** [review]

**Pull request merged**